sys/net/gnrc/link_layer/lorawan/include/gnrc_lorawan_internal.h

```diff
@@ -137,13 +141,23 @@ typedef struct {
     uint8_t num_gateways;   /**< number of gateways */
 } mlme_link_req_confirm_t;
 
+/**
+ * @brief MLME Link ADR request data sent by NS
+ */
+typedef struct {
+    uint8_t dr_txpwr;       /**< DataRate_TXPower (in dB) */
+    uint16_t chmsk;         /**< Channel Mask*/
+    uint8_t redncy;         /**< Redundancy bit */
+} mlme_link_adr_req_t;
+
 /**
  * @brief MCPS data
  */
 typedef struct {
     iolist_t *pkt;          /**< packet of the request */
     uint8_t port;           /**< port of the request */
     uint8_t dr;             /**< datarate of the request */
+    bool adr;               /**< adr bit (FCtrl) status */
```

Comment:
as mentioned above, this doesn't belong here but rather to `gnrc_lorawan_t`.

Reply:
I think it would make more sense to have it in `gnrc_lorawan_mlme_t`.

Reply:
yes, `gnrc_lorawan_mlme_t`.